packages/core/src/tests/root.test.ts

```diff
@@ -0,0 +1,47 @@
+import {createRemoteRoot} from '../root';
+
+describe('root', () => {
+  it('does not throw error when no allowed components are set', () => {
```

Comment:
I'd prefer if these tests were scoped under a `describe('createComponent')`, and we had additional tests under nested `describe('appendChild()')` and `describe('insertChildAfter')` blocks to check the "appending a node not created by this root" validation.

Reply:
Fixed!